crates/cli/src/main.rs

```diff
@@ -133,15 +190,53 @@ fn set_hostname_if_missing(host: &str, extra_options: &mut Option<ExtraRequestSe
 /// # Arguments
 /// * `args` - A reference to the command line options.
 /// * `result` - A reference to the result of the query.
-fn output_result(args: &Cli, result: &dyn CommonResponse) {
-    match args.output_mode {
+fn output_result<T: CommonResponse + ?Sized>(output_mode: OutputMode, format: OutputFormat, result: &T) {
+    match format {
+        OutputFormat::Debug => {
+            match output_mode {
+                OutputMode::Generic => output_result_debug(result.as_json()),
+                OutputMode::ProtocolSpecific => output_result_debug(result.as_original()),
+            }
+        }
         #[cfg(feature = "json")]
-        OutputMode::Generic if args.json => output_result_json(result.as_json()),
+        OutputFormat::JsonPretty => {
+            match output_mode {
+                OutputMode::Generic => output_result_json_pretty(result.as_json()),
+                OutputMode::ProtocolSpecific => output_result_json_pretty(result.as_original()),
+            }
+        }
         #[cfg(feature = "json")]
-        OutputMode::ProtocolSpecific if args.json => output_result_json(result.as_original()),
-
-        OutputMode::Generic => output_result_debug(result.as_json()),
-        OutputMode::ProtocolSpecific => output_result_debug(result.as_original()),
+        OutputFormat::Json => {
+            match output_mode {
+                OutputMode::Generic => output_result_json(result.as_json()),
+                OutputMode::ProtocolSpecific => output_result_json(result.as_original()),
+            }
+        }
+        #[cfg(feature = "xml")]
+        OutputFormat::Xml => {
+            match output_mode {
+                OutputMode::Generic => output_result_xml(result.as_json()),
```

Comment:
I got this error when trying to use XML with TF2 and generic output:

```
called `Result::unwrap()` on an `Err` value: Writer { source: LastElementNameNotAvailable }
```

Reply:
can you send the command used if possible to reproduce, thanks

Reply:
on this note also i think that we may need tests for the cli

Reply:
The command I used was: `cargo run -p gamedig-cli -- query -g teamfortress2 -i 193.221.192.41 -f xml -o generic`

Reply:
i looked into this and our imitation was the xml lib that was being used. I got a fix incoming where i convert to json and parse the json to xml using different type of xml lib, bit of a pain but this should cover all cases. I tested this command above and a few others with no issues
![image](https://github.com/gamedig/rust-gamedig/assets/75994858/0e865bd0-f58f-4fe6-92e6-7c9918c2bfae)